var/spack/repos/builtin/packages/dav-sdk/package.py

```diff
@@ -0,0 +1,177 @@
+
+    # HDF5 VOL Adapters require hdf5@1.14:
+
+    # hdf5-vfd-gds needs cuda@11.7.1 or later, only enable when 11.7.1+ available.
+    depends_on("hdf5-vfd-gds@1.0.2:", when="+cuda+hdf5 ^cuda@11.7.1: ^hdf5@1.14:")
+    for cuda_arch in cuda_arch_variants:
+        depends_on(
+            "hdf5-vfd-gds@1.0.2: {0}".format(cuda_arch),
+            when="+cuda+hdf5 {0} ^cuda@11.7.1: ^hdf5@1.14:".format(cuda_arch),
+        )
+    conflicts("~cuda", when="^hdf5-vfd-gds@1.0.2:")
+    conflicts("~hdf5", when="^hdf5-vfd-gds@1.0.2:")
+    conflicts("~hdf5", when="^hdf5-vol-async")
+    conflicts("~hdf5", when="^hdf5-vol-cache")
+    conflicts("~hdf5", when="^hdf5-vol-log")
+    depends_on("hdf5-vol-async", when="+hdf5 ^hdf5@1.14:")
+    depends_on("hdf5-vol-cache", when="+hdf5 ^hdf5@1.14:")
+    depends_on("hdf5-vol-log", when="+hdf5 ^hdf5@1.14:")
+
+    dav_sdk_depends_on("parallel-netcdf+shared", when="+pnetcdf", propagate=["fortran"])
+
+    # Fortran support with ascent is problematic on some Cray platforms so the
+    # SDK is explicitly disabling it until the issues are resolved.
+    dav_sdk_depends_on(
+        "ascent+mpi~fortran+python+shared+vtkh+dray~test",
+        when="+ascent",
+        propagate=["adios2", "cuda"] + cuda_arch_variants,
+    )
+    depends_on("ascent+openmp", when="~rocm+ascent")
+    depends_on("ascent~openmp", when="+rocm+ascent")
+
+    # Need to explicitly turn off conduit hdf5_compat in order to build
+    # hdf5@1.12 which is required for SDK
+    depends_on("conduit ~hdf5_compat", when="+ascent +hdf5")
+    # Disable configuring with @develop. This should be removed after ascent
+    # releases 0.8 and ascent can build with conduit@0.8: and vtk-m@1.7:
+    conflicts("^ascent@develop", when="+ascent")
+
+    dav_sdk_depends_on("diy@3:", when="+diy")
+
+    # ParaView needs @5.11: in order to use CUDA/ROCM, therefore it is the minimum
+    # required version since GPU capability is desired for ECP
+    dav_sdk_depends_on(
+        "paraview@5.11:+mpi+openpmd+python+kits+shared+catalyst+libcatalyst" " use_vtkm=on",
+        when="+paraview",
+        propagate=["adios2", "cuda", "hdf5", "rocm"] + amdgpu_target_variants + cuda_arch_variants,
+    )
+    dav_sdk_depends_on("libcatalyst@2:+mpi", when="+paraview")
+    conflicts("^paraview@master", when="+paraview")
+
+    dav_sdk_depends_on("visit+mpi+python+silo", when="+visit", propagate=["hdf5", "adios2"])
+
+    dav_sdk_depends_on(
+        "vtk-m@1.7:+shared+mpi+rendering",
+        when="+vtkm",
+        propagate=["cuda", "rocm"] + cuda_arch_variants + amdgpu_target_variants,
+    )
+    # TODO: When Ascent is updated to use VTK-m >= 1.8 move examples to
+    # the main spec.
+    conflicts("^vtk-m~examples", when="+vtkm ^vtk-m@1.8:")
+    depends_on("vtk-m+openmp", when="~rocm+vtkm")
+    depends_on("vtk-m~openmp", when="+rocm+vtkm")
+
+    dav_sdk_depends_on("zfp", when="+zfp", propagate=["cuda"] + cuda_arch_variants)
```

Comment:
adios2 depends in zfp

Reply:
This is adding ZFP to the DAV packages. Adios2 handles the ZFP extension using the DAV variant propagation logic.